src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int stSize = 10;
+    private int realSize = 0;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[stSize];
+    }
+
+    private void resize() {
+        if (realSize >= stSize) {
+            stSize *= 1.5;
+            T[] newItems = (T[]) new Object[stSize];
+            System.arraycopy(items, 0, newItems, 0, realSize);
+            items = newItems;
+        }
+    }
+
+    private void testIndexOk(int index) {
```

Comment:
Bad name for the method. 
```suggestion
    private void checkIndex(int index) {
```